pyface/data_view/i_data_view_widget.py

```diff
@@ -8,52 +8,283 @@
 #
 # Thanks for using Enthought open source!
 
-from traits.api import Bool, HasStrictTraits, Instance
+from contextlib import contextmanager
+import logging
+
+from traits.api import (
+    Bool, ComparisonMode, Enum, HasStrictTraits, Instance, List, Property,
+    TraitError, Tuple, cached_property,
+)
 
 from pyface.data_view.abstract_data_model import AbstractDataModel
 from pyface.i_widget import IWidget
 
 
+logger = logging.getLogger(__name__)
+logger.level = logging.DEBUG
+
+
 class IDataViewWidget(IWidget):
+    """ Interface for data view widgets. """
+
+    #: The data model for the data view.
+    data_model = Instance(AbstractDataModel, allow_none=False)
 
     #: Whether or not the column headers are visible.
     header_visible = Bool(True)
 
-    #: The data model for the data view.
-    data_model = Instance(AbstractDataModel, allow_none=False)
+    #: What can be selected.  Some backends may support more.
+    selection_type = Enum("row",)
+
```

Comment:
Strong disagreement on this.  Listing all possible values here doesn't help with testing if the _behaviour_ is different.  If I set the `selection_type` to `"column"` and a toolkit might ignore that and goes with the default `"row"` how can I test it reliably?  Remember that a toolkit might not even be implemented in Pyface, so it can't add a "skip if this is my toolkit" to shared tests.

In fact everywhere that we have a skip based on a particular toolkit value is a potential problem for someone implementing a stand-alone toolkit.

This is leading me to thing that I should be moving all of the tests which are effectively Qt only into a separate qt-specific test suite.

Or alternatively, we need a way of advertising capabilities (perhaps by introspecting the appropriate traits?)